### DISCUSSION TITLE
Suggestion for the ability to delete a message if a user sends the wrong link

### DISCUSSION THREAD

**Discussion by Lqkuq** [everettsouthwick/embetter#2] [Ideas]
Hello, your bot is very useful!
I don't know if you are open to feature suggestions, but I would like to write about something that I think would be useful to have.
It would be nice to be able to generate a button in the bot's message after the link has been replaced, so that the user who sent the wrong link can delete the message.
Best regards

**Comment by everettsouthwick** [comment]
Hello,

Thank you very much for the kind comment! I'm very open to feature suggestions if there is anything else you think would be good to add. 

I will definitely look into adding a button to delete the senders message if in `REPLY` mode. I may also investigate the feasibility of adding an embed button if running in `MANUAL` mode to allow that shortcut.

Thanks again!

**Reply by Lqkuq** [reply]
Thanks for the reply.
there are no other features I can think of at this time. Looking forward to adding the feature!

**Comment by everettsouthwick** [comment]
This has been implemented in #3. I will need to also add it to the slash command /embed workflow, but the more used workflow of people posting links has this feature now.

**Comment by Lqkuq** [comment]
Thanks for taking my feature suggestion! But sorry, maybe you misunderstood.

I was thinking that it would be nice if we could delete the bot messages instead of the user's original message.
Users are human and sometimes they make mistakes in the links they post, but currently only a moderator can delete a bot message. In that case, the wrong post will remain on the channel unless it is deleted by a moderator.

This leads to more work for the moderators, which seems a bit tedious.

**Reply by everettsouthwick** [reply]
Ah, that makes sense! I will change the logic to allow you to delete the bots message.

**Reply by Lqkuq** [reply]
Sorry for the missing explanation in my first post!

**Reply by everettsouthwick** [reply]
No worries at all, I got the wrong idea. Your idea makes more sense from a moderation perspective, so I think that's the better approach. I just implemented this in #6 to allow the bot message to be deleted

**Reply by Lqkuq** [reply]
Thanks for the quick response!
I hope your bot will be used on various servers.
